db/migrate/20200622222120_copy_page_intro_to_new_header_textbox.rb

```diff
@@ -0,0 +1,67 @@
+class CopyPageIntroToNewHeaderTextbox < ActiveRecord::Migration
+  module Embeddable
+    class Embeddable::Xhtml < ActiveRecord::Base
+      attr_accessible :name, :content, :is_hidden, :is_full_width
+      has_many :page_items, :as => :embeddable, :dependent => :destroy
+      has_many :interactive_pages, :through => :page_items
+    end
+
+    def self.table_name_prefix
+      'embeddable_'
+    end
+  end
+
+  class InteractivePage < ActiveRecord::Base
+    has_many :page_items, :order => [:section, :position]
+  end
+
+  class PageItem < ActiveRecord::Base
+    attr_accessible :interactive_page, :position, :section, :embeddable,
+                    :interactive_page_id, :embeddable_id, :embeddable_type
+    acts_as_list :scope => :interactive_page
+    belongs_to :interactive_page
+    belongs_to :embeddable, :polymorphic => true, dependent: :destroy
+  end
+
+  def up
+    InteractivePage.find_each(batch_size: 10) do |ip|
+      if ip.text.present?
+        ip.show_header = true
```

Comment:
This should set show_header to the value of show_introduction. I suspect there are some production activity pages which have hidden introductions but still have some text that the author never cleaned up.